.github/workflows/build-website.yml

```diff
@@ -21,3 +23,34 @@ jobs:
       - name: Build website
         run: |
           npm run build
+      - name: Upload html output as artifact
+        uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 # v5.0.0
+        with:
+          name: html-${{ github.run_id }}
+          path: ./dist
+  link_check:
+    name: Check internal links
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout lychee config file from repository
+        uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8 # v6.0.1
```

Comment:
This just updated the checkout action to a hash so it's pinned and the current version! let's merge!